File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/exceptions/VerityException.java
package com.evernym.verity.sdk.exceptions;

/**
 * Generic and abstract exception class for exceptions throw by the Verity-SDK
 */
public abstract class VerityException extends Exception {
    /**
     * Constructs a new VerityException with the given message.
     *
     * @param message the given message for the cause of the exception.
     */
    public VerityException(String message) {super(message);}

    /**
     * Constructs a new VerityException with the given message and cause.
     *
     * @param message the given message for the cause of the exception.
     * @param cause the given cause (Throwable) for the exception.
     */
    public VerityException(String message, Throwable cause) {super(message, cause);}
}

File: sdk/dotnet-sdk/src/VeritySDK/Exceptions/VerityException.cs
using System;

namespace VeritySDK.Exceptions
{
    /// <summary>
    /// Generic and abstract exception class for exceptions throw by the Verity-SDK
    /// </summary>
    public abstract class VerityException : System.Exception
    {
        /// <summary>
        /// Constructs a new with the given message.
        /// </summary>
        /// <param name="message">message the given message for the cause of the exception.</param>
        public VerityException(string message) : base(message) { }

        /// <summary>
        /// Constructs a new with the given message and cause.
        /// </summary>
        /// <param name="message">the given message for the cause of the exception.</param>
        /// <param name="cause">the given cause for the exception.</param>
        public VerityException(String message, System.Exception cause) : base(message, cause) { }
    }
}

